src/Modules/ASayModule/Services/ASayService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using EvoSC.Common.Interfaces.Controllers;
+using EvoSC.Common.Interfaces.Localization;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.ASayModule.Events;
+using EvoSC.Modules.Official.ASayModule.Interfaces;
+using Microsoft.Extensions.Logging;
+
+namespace EvoSC.Modules.Official.ASayModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Transient)]
+public class ASayService : IASayService
+{
+    private readonly ILogger<ASayService> _logger;
+    private readonly IManialinkManager _manialinkManager;
+    private readonly IContextService _contextService;
+    private readonly dynamic _locale;
+
+    public ASayService(ILogger<ASayService> logger, IManialinkManager manialinkManager, IContextService context, Locale locale)
+    {
+        _logger = logger;
+        _manialinkManager = manialinkManager;
+        _contextService = context;
+        _locale = locale;
+    }
+
+    public async Task OnDisableAsync()
+    {
+        await _manialinkManager.HideManialinkAsync("ASayModule.Announcement");
```

Comment:
This is actually not necessary as the module system will automatically hide all manialinks from the module when it is disabled.

Reply:
removed this method entirely